src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixMult.java

```diff
@@ -123,6 +126,52 @@ public static void matrixMult(MatrixBlock m1, MatrixBlock m2, MatrixBlock ret, i
 	{
 		matrixMult(m1, m2, ret, rl, ru, true);
 	}
+	
+	// We could encapsulate heuristics in this function
+	// For now, we only consider matrix-vector operation to be memory bound
+	private static boolean isMatMultMemoryBound(int m1Rlen, int m1Clen, int m2Clen) {
+		if(m1Rlen == 1 || m1Clen == 1 || m2Clen == 1)
+			return true;
+		else
+			return false;
```

Comment:
please globally remove such if-else statements and replace with "return (condition);"